drivers/sensor/st/stm32_digi_temp/stm32_digi_temp.c

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2024 Aurelien Jarno
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_digi_temp
+
+#include <zephyr/device.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(stm32_digi_temp, CONFIG_SENSOR_LOG_LEVEL);
+
+/* Constants */
+#define ONE_MHZ			1000000 /* Hz */
+#define TS1_T0_VAL0		30	/* °C */
+#define TS1_T0_VAL1		130	/* °C */
+#define SAMPLING_TIME		15	/* best precision */
+
+struct stm32_digi_temp_data {
+	struct k_sem sem_isr;
+	struct k_mutex mutex;
+
+	/* Peripheral clock frequency */
+	uint32_t pclk_freq;
+	/* Engineering value of the frequency measured at T0 in Hz */
+	uint32_t t0_freq;
+	/* Engineering value of the T0 temperature in °C */
+	uint16_t t0;
+	/* Engineering value of the ramp coefficient in Hz / °C */
+	uint16_t ramp_coeff;
+
+	/* Raw sensor value */
+	uint16_t raw;
+};
+
+struct stm32_digi_temp_config {
+	/* DTS instance. */
+	DTS_TypeDef *base;
+	/* Clock configuration. */
+	struct stm32_pclken pclken;
+};
+
+static void stm32_digi_temp_isr(const struct device *dev)
+{
+	struct stm32_digi_temp_data *data = dev->data;
+	const struct stm32_digi_temp_config *cfg = dev->config;
+
+	/* Clear interrupt */
+	cfg->base->ICIFR |= DTS_ICIFR_TS1_CITEF;
```

Comment:
Did you try using the stm32h7xx_hal_dts functions like __HAL_DTS_CLEAR_FLAG ?

Reply:
Nope I haven't tried that, I was really looking for a LL header, and therefore didn't look at that one in details. I will give it a try.

Reply:
@FRASTM This doesn't work as this macro takes a `DTS_HandleTypeDef` handle as parameter, which is the HAL specific object on which the `HAL_DTS_*` methods act.

Reply:
OTOH, I have just pushed a new version using `CLEAR_BIT`, `SET_BIT` and `MODIFY_REG`.